plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -202,7 +202,7 @@ private void evaluateFacetConfiguration() {
   private void updatePageBasedOnFacetConfig(boolean hasCorrectFacetConfiguration) {
     if (hasCorrectFacetConfiguration) {
       canSetMessage = true;
-      setErrorMessage(null);
+      content.getDataBindingContext().updateTargets();
```

Comment:
Further update on this: the bug referenced in https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/880#issuecomment-256149028 is a gtk3 bug. If we run the plugin at the commit just prior to the comment and open the deploy preferences page, then click the `override version` checkbox (to set it), we expect to see an error message at the top of the page, that the version number is invalid (because it is an empty string currently). Based on the GTK version we see:
 - GTK2 will display the error message
 - GTK3 will not display the error message

So apparently PR #880 worked around the GTK3 bug without realizing that it was the underlying issue. I'm investigating which commit was the accidental workaround.